ctk/common/src/test/kotlin/org/codice/compilance/verification/core/SamlDefinedIdentifiersVerifierSpec.kt

```diff
@@ -0,0 +1,281 @@
+import org.codice.compliance.SAMLCore_8_2_3_a
+import org.codice.compliance.SAMLCore_8_3_2_a
+import org.codice.compliance.SAMLCore_8_3_6_a
+import org.codice.compliance.SAMLCore_8_3_6_b
+import org.codice.compliance.utils.TestCommon.Companion.ASSERTION_NAMESPACE
+import org.codice.compliance.utils.TestCommon.Companion.ENTITY
+import org.codice.compliance.utils.TestCommon.Companion.PROTOCOL_NAMESPACE
+import org.codice.compliance.utils.TestCommon.Companion.VERSION
+import org.codice.compliance.verification.core.SamlDefinedIdentifiersVerifier
+import org.codice.compliance.verification.core.SamlDefinedIdentifiersVerifier.Companion.ATTRIBUTE_NAME_FORMAT_BASIC
+import org.codice.compliance.verification.core.SamlDefinedIdentifiersVerifier.Companion.ATTRIBUTE_NAME_FORMAT_UNSPECIFIED
+import org.codice.compliance.verification.core.SamlDefinedIdentifiersVerifier.Companion.ATTRIBUTE_NAME_FORMAT_URI
+import org.codice.compliance.verification.core.SamlDefinedIdentifiersVerifier.Companion.ENTITY_ID_MAX_LEN
+import org.codice.compliance.verification.core.SamlDefinedIdentifiersVerifier.Companion.NAME_ID_FORMAT_EMAIL
+import org.w3c.dom.Node
+import java.time.Instant
+
+class SamlDefinedIdentifiersVerifierSpec : StringSpec() {
+    init {
+        val validEntityId = "ValidEntityID"
+        val maxLengthEntityId = "A".repeat(ENTITY_ID_MAX_LEN)
+
+        val now = Instant.now()
+
+        @Suppress("LongParameterList")
+        fun createResponse(attributeName: String = "Unspecified",
+                           attributeFormat: String = ATTRIBUTE_NAME_FORMAT_UNSPECIFIED,
+                           identifierValue: String = "example-email@domain.com",
+                           identifierFormat: String = NAME_ID_FORMAT_EMAIL,
+                           extraIdentifierAttribute: String = ""): Node {
+            return buildDom(
+                    """
+                    |<samlp:Response
+                    |  xmlns:saml="$ASSERTION_NAMESPACE"
+                    |  xmlns:samlp="$PROTOCOL_NAMESPACE"
+                    |  ID="id"
+                    |  Version="$VERSION"
+                    |  IssueInstant="$now">
+                    |  <samlp:Status>
+                    |    <samlp:StatusCode Value="urn:oasis:names:tc:SAML:2.0:status:Success"/>
+                    |  </samlp:Status>
+                    |  <saml:Assertion ID="id" IssueInstant="$now" Version="$VERSION">
+                    |    <saml:Issuer
+                    |    $extraIdentifierAttribute
+                    |    Format="$identifierFormat">$identifierValue</saml:Issuer>
+                    |    <saml:AttributeStatement>
+                    |      <saml:Attribute Name="$attributeName" NameFormat="$attributeFormat" />
+                    |    </saml:AttributeStatement>
+                    |  </saml:Assertion>
+                    |</samlp:Response>
+                    """.trimMargin())
+        }
+
+        /* 8.2 URI/Basic name attribute formats */
```

Comment:
✏️ Do we also want to test when either the name or nameFormat attributes are `null`?

Reply:
Implementing now.

Note that `NameFormat` cannot be null by Schema Validation, so that is not going to be implemented.